demo/contractHost/vat-fred.js

```diff
@@ -19,85 +19,36 @@ function makeFredMaker(E, host, log) {
-      const inviteIssuerLabel = harden({
-        issuer: inviteIssuerP,
-        description: 'contract host',
-      });
-      const moneyIssuerP = E(myMoneyPurseP).getIssuer();
-      const stockIssuerP = E(myStockPurseP).getIssuer();
-      const finIssuerP = E(myFinPurseP).getIssuer();
+      const dough10P = E(E(myMoneyPurseP).getIssuer()).makeAmount(10);
+      const wonka7P = E(E(myStockPurseP).getIssuer()).makeAmount(7);
+      const fin55P = E(E(myFinPurseP).getIssuer()).makeAmount(55);
 
       const fred = harden({
         acceptOptionOffer(allegedSaleInvitePaymentP) {
           log('++ fred.acceptOptionOffer starting');
 
-          const dough10 = harden({
-            label: {
-              issuer: moneyIssuerP,
-              description: 'dough',
-            },
-            quantity: 10,
-          });
-          const wonka7 = harden({
-            label: {
-              issuer: stockIssuerP,
-              description: 'wonka',
-            },
-            quantity: 7,
-          });
-          const fin55 = harden({
-            label: {
-              issuer: finIssuerP,
-              description: 'fins',
-            },
-            quantity: 55,
-          });
-
-          const allegedSaleAmountP = E(allegedSaleInvitePaymentP).getBalance();
-
-          const verifiedSaleInvitePaymentP = E.resolve(allegedSaleAmountP).then(
-            allegedSaleInviteAmount => {
-              const allegedOptionsInviteAmount =
-                allegedSaleInviteAmount.quantity.terms.right;
-
-              const optionsInviteAmount = harden({
-                label: inviteIssuerLabel,
-                quantity: {
-                  installation: coveredCallInstallationP,
-                  terms: [
-                    escrowExchangeInstallationP,
-                    dough10,
-                    wonka7,
-                    timerP,
-                    'singularity',
-                  ],
-                  seatIdentity:
-                    allegedOptionsInviteAmount.quantity.seatIdentity,
-                  seatDesc: 'holder',
-                },
-              });
-
-              const saleInviteAmountP = allComparable(
-                harden({
-                  label: inviteIssuerLabel,
-                  quantity: {
-                    installation: escrowExchangeInstallationP,
-                    terms: { left: fin55, right: optionsInviteAmount },
-                    seatIdentity: allegedSaleInviteAmount.quantity.seatIdentity,
-                    seatDesc: 'left',
-                  },
-                }),
+          const coveredCallTermsP = [dough10P, wonka7P, timerP, 'singularity'];
+          const verifiedSaleInvitePaymentP = E(allegedSaleInvitePaymentP)
+            .getBalance()
+            .then(balance => {
+              return E.resolve(fin55P).then(f55 =>
+                E(escrowExchangeInstallationP)
+                  .checkPartialAmount(balance, f55)
```

Comment:
Yeah, this one. How does fred know whether he's just verified that the left or the right is for 55 fin?